tests/scripts/all.sh

```diff
@@ -2930,26 +2958,21 @@ psa_crypto_config_accel_all_curves_except_one () {
     scripts/config.py unset MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED
     scripts/config.py unset MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED
 
-    # Explicitly disable all SW implementation for elliptic curves
-    scripts/config.py unset-all "MBEDTLS_ECP_DP_[0-9A-Z_a-z]*_ENABLED"
-    # Just leave SW implementation for the specified curve for allowing to
-    # build with ECP_C.
-    scripts/config.py set $BUILTIN_CURVE
+    # Explicitly disable all SW implementation for elliptic curves, keeping
+    # only the requested one.
+    helper_disable_builtin_curves $BUILTIN_CURVE
     # Accelerate all curves listed in "crypto_config.h" (skipping the ones that
     # are commented out).
     # Note: Those are handled in a special way by the libtestdriver machinery,
     # so we only want to include them in the accel list when building the main
     # libraries, hence the use of a separate variable.
-    loc_curve_list=""
-    for CURVE in $(sed -n 's/^#define PSA_WANT_\(ECC_[0-9A-Z_a-z]*\).*/\1/p' <"$CRYPTO_CONFIG_H"); do
-        loc_curve_list="$loc_curve_list $CURVE"
-    done
+    loc_curve_list="$(helper_get_psa_curve_list)"
 
     # Build
     # -----
 
     # These hashes are needed for some ECDSA signature tests.
-    loc_extra_list="ALG_SHA_224 ALG_SHA_256 ALG_SHA_384 ALG_SHA_512"
+    loc_extra_list="ALG_SHA_1 ALG_SHA_224 ALG_SHA_256 ALG_SHA_384 ALG_SHA_512"
     helper_libtestdriver1_make_drivers "$loc_accel_list" "$loc_extra_list"
 
     # (See above regarding loc_curve_list.)
```

Comment:
Here and in several other components that are [failing as of c5fa53fc53348571bc3e49503bf530bd31eef5c5](https://mbedtls.trustedfirmware.org//job/mbed-tls-pr-head/job/PR-8075-head/21/flowGraphTable/), I don't understand the logic of the test. Since there is one curve that isn't accelerated, and ECDH (etc.) is requested, `ecdh.c` does get built. The logic of the library makes sense. Why does the test want the built-in ECDH not to be enabled even though it sets up a configuration that requires it?

Reply:
Several tests date back from a time when we were not reasoning correctly about acceleration and possibly rely on flawed logic. I'll review them and either fix them, or I guess even remove a few that might not make sense any more. (That was next on my list, and is the main reason I haven't re-added the label "needs-review" yet.)